packages/vue-renderer/src/renderer.js

```diff
@@ -418,6 +418,9 @@ export default class VueRenderer {
     // Call ssr:csp hook
     await this.context.nuxt.callHook('vue-renderer:ssr:csp', cspScriptSrcHashes)
 
+    // Add csp meta tags
+    HEAD += `<meta http-equiv="Content-Security-Policy" content="script-src ${cspScriptSrcHashes.join()}">`
```

Comment:
I'd guard that and checka whether CSP is enabled or not.

Also, I might would hide that behind a feature flag 🤔 	
And maybe only enable it "by default" (if CSP is enabled/set) on nuxt generate
(cc @pi0)

Reply:
Nice point @manniL. I agree

Reply:
@manniL @pi0 I didn't get you :confused: 
Can you just have a mention no how should it be changed :thinking:

Reply:
@manniL @pi0 I didn't get you :confused: 
Can you just have a mention on how should it be changed :thinking:

Reply:
@jamesgeorge007 	No worries ☺️ 	

1. Check whether the user actually set sth. in the config's [render.csp](https://nuxtjs.org/api/configuration-render#csp)
2. We should use another config value to only enable that feature if the user *explicitly* enables it through the config. It could be also inside the render.csp object

Reply:
I guess wrapping  it within ```js if (this.context.options.csp) ``` checks whether `csp` is enabled and acts appropriately. Does this makes sense to you?
 I'm not sure on what is to be done further :thinking:

Reply:
I guess wrapping  it within ``if (this.context.options.csp)` checks whether `csp` is enabled and acts appropriately. Does this makes sense to you?
 I'm not sure on what is to be done further :thinking:

Reply:
@manniL I guess wrapping  it within ``if (this.context.options.csp)` checks whether `csp` is enabled and acts appropriately. Does this make sense to you?
 I'm not sure on what is to be done further :thinking:

Reply:
I guess wrapping  it within `if (this.context.options.csp)` checks whether `csp` is enabled and acts appropriately. Does this make sense to you?
 I'm not sure on what is to be done further :thinking: